stablehlo/reference/Ops.cpp

```diff
@@ -102,6 +102,50 @@ Tensor evalClampOp(const Tensor &min, const Tensor &operand, const Tensor &max,
   return result;
 }
 
+Tensor evalCompareOp(const Tensor &lhs, const Tensor &rhs,
+                     ComparisonDirection comparisonDirection,
+                     TensorType resultType) {
+  Tensor result(resultType);
+  auto elementTy = lhs.getElementType();
+
+  if (isSupportedComplexType(elementTy) &&
+      (comparisonDirection == ComparisonDirection::GE ||
+       comparisonDirection == ComparisonDirection::GT ||
+       comparisonDirection == ComparisonDirection::LE ||
+       comparisonDirection == ComparisonDirection::LT)) {
+    report_fatal_error(invalidArgument(
+        "Unsupported element type %s for comparison direction %s",
+        debugString(elementTy).c_str(),
+        debugString(comparisonDirection).c_str()));
+  }
+
+  for (auto it = result.index_begin(); it != result.index_end(); ++it) {
+    bool cmpResult = false;
+    switch (comparisonDirection) {
+      case ComparisonDirection::EQ:
+        cmpResult = lhs.get(*it) == rhs.get(*it);
+        break;
+      case ComparisonDirection::NE:
+        cmpResult = lhs.get(*it) != rhs.get(*it);
+        break;
+      case ComparisonDirection::GE:
+        cmpResult = lhs.get(*it) >= rhs.get(*it);
+        break;
+      case ComparisonDirection::GT:
+        cmpResult = lhs.get(*it) > rhs.get(*it);
+        break;
+      case ComparisonDirection::LE:
+        cmpResult = lhs.get(*it) <= rhs.get(*it);
+        break;
+      case ComparisonDirection::LT:
+        cmpResult = lhs.get(*it) < rhs.get(*it);
+        break;
```

Comment:
Let's insert a `default` case here, so that this doesn't start to silently misbehave if we add another comparison direction.